src/code_804267C.c

```diff
@@ -0,0 +1,320 @@
+#include "global.h"
+#include "code_804267C.h"
+#include "code_8045A00.h"
+#include "code_806CD90.h"
+#include "dungeon.h"
+#include "dungeon_engine.h"
+#include "dungeon_leader.h"
+#include "dungeon_music.h"
+#include "dungeon_util.h"
+#include "structs/str_dungeon.h"
+
+extern u8 gAvailablePokemonNames[]; // 202DF98
+
+extern const u16 gUnknown_80F67DC[4];
+extern const u16 gUnknown_80F67E4[4];
+extern const s32 gUnknown_80F67EC[8];
+extern const u32 gUnknown_80F680C[8];
+extern const u16 gUnknown_80F682C[8];
+extern const u8 *gPtrFeralFoundItemMessage[];
+
+extern void sub_803E46C(u32);
+extern void sub_803E708(u32 r0, u32 r1);
+extern void sub_8040238(void);
+extern void sub_804151C(Entity *pokemon, u32 r1, u8 r2);
+extern void sub_804178C(u32);
+extern void sub_80416A4(Position *r0, u32 r1, bool8 r2);
+extern void sub_80421C0(Entity *pokemon, u16 r1);
+extern void sub_80429A0(Entity *);
+extern void sub_8042B34(u32, u32, u32);
+extern bool8 sub_8042CC0(void);
+extern void sub_8042D7C(void);
+extern u8 sub_8043D10(void);
+extern bool8 sub_8045888(Entity *);
+extern void sub_8052210(u32);
+extern void sub_805239C(Entity *, const u8 *);
+
+// This func plays the appropriate ascend/descend SFX for stairs depending on the direction
+void PlayStairsSound(void)
+{
+    u32 songIndex;
+
+    songIndex = 997;
+
+    switch (sub_8043D10()) {
+        case 0:
+        default:
+            songIndex = IsStairDirectionUp(gDungeon->dungeonLocation.id) ? 340 : 341;
+            break;
+        case 1:
+        case 2:
+            break;
+    }
+
+    PlaySoundEffect(songIndex);
+}
+
+void sub_80426C8(u32 a0, u32 a1)
+{
+    if (a1 == 3)
+        DungeonStartNewBGM(0);
+
+    if (a0 != 4)
+        PlaySoundEffect(gUnknown_80F67DC[a1]);
+    else
+        PlaySoundEffect(gUnknown_80F67E4[a1]);
+
+    sub_8042B34(a0, a1, 0x80);
+
+    while (sub_8042CC0())
+        sub_803E46C(0x46);
+
+    sub_8042D7C();
+    sub_8040238();
+    sub_8052210(1);
+}
+
+void sub_8042730(Entity *a0, void *unused)
+{
+    s32 i;
+
+    sub_806CDD4(a0, 8, 8);
+    for (i = 0; i < 100; i++) {
+        sub_803E46C(0x51);
+        if ((a0->spriteInfo.unk18 & 3) != 0 || a0->unk6B != 8)
+            break;
+    }
+}
+
+bool8 sub_8042768(Entity* pokemon)
+{
+    if (!pokemon->isVisible)
+        return FALSE;
+    if (gDungeon->blinded) {
+        if (gDungeon->cameraTarget == pokemon)
+            return TRUE;
+        else
+            return FALSE;
+    }
+    return sub_8045888(pokemon);
+}
+
+void sub_80427AC(void)
+{
+    Entity *entity;
+    EntityInfo *enInfo;
+    s32 i;
+
+    for (i = 0; i < 20; i++) {
+        entity = gDungeon->allPokemon[i];
+        if (EntityExists(entity)) {
+            enInfo = entity->info;
+            if (enInfo->fillF3) {
```

Comment:
fillF3 should be named

Reply:
Any ideas what this variable is for? Otherwise I can use `unkF3` to indicate that it's used rather than filler.

Reply:
Going ahead with `unkF3`.